biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -816,7 +768,147 @@ subroutine terminate_cohorts( currentSite, currentPatch, level )
+    
+    ! -----------------------------------------------------------------------------------
+    ! This routine transfers the existing mass in all pools and all elements
+    ! on a vegetation cohort, into the litter pool.
+    ! 
+    ! Important: (1) This IS NOT turnover, this is not a partial transfer.
+    !            (2) This is from a select number of the cohort
+    !            (3) This does not affect the PER PLANT mass pools, so 
+    !                do not update any PARTEH structures.
+    !            (4) The change in plant number density (due to death or termination)
+    !                IS NOT handled here.
+    !            (5) This routine is NOT used for disturbance, mostly
+    !                because this routine assumes a cohort lands in its patch
+    !                Whereas the disturbance scheme does NOT assume that.
+    ! -----------------------------------------------------------------------------------
+
+    ! Arguments
+    type (ed_site_type)   , target  :: csite
+    type (ed_patch_type)  , target  :: cpatch
+    type (ed_cohort_type) , target  :: ccohort
+    real(r8)                        :: nplant     ! Number (absolute)
+                                                  ! of plants to transfer
+    
+    !
+    type(litter_type), pointer        :: litt       ! Litter object for each element
+    type(site_fluxdiags_type),pointer :: flux_diags
+
+    real(r8) :: leaf_m    ! leaf mass [kg]
+    real(r8) :: store_m   ! storage mass [kg]
+    real(r8) :: sapw_m    ! sapwood mass [kg]
+    real(r8) :: fnrt_m    ! fineroot mass [kg]
+    real(r8) :: repro_m   ! reproductive mass [kg]
+    real(r8) :: struct_m  ! structural mass [kg]
+    real(r8) :: plant_dens! plant density [/m2]
+    real(r8) :: dcmpy_frac! fraction of mass going to each decomposability partition
+    integer  :: el        ! loop index for elements
+    integer  :: c         ! loop index for CWD
+    integer  :: pft       ! pft index of the cohort
+    integer  :: sl        ! loop index for soil layers
+    integer  :: dcmpy     ! loop index for decomposability
+    
+    !----------------------------------------------------------------------
+
+    pft = ccohort%pft
+
+    plant_dens = nplant/cpatch%area
+
+    call set_root_fraction(csite%rootfrac_scr, pft, csite%zi_soil, &
+         icontext = i_biomass_rootprof_context)
+
+    do el=1,num_elements
+       
+       leaf_m   = ccohort%prt%GetState(leaf_organ, element_list(el))
+       store_m  = ccohort%prt%GetState(store_organ, element_list(el))
+       sapw_m   = ccohort%prt%GetState(sapw_organ, element_list(el))
+       fnrt_m   = ccohort%prt%GetState(fnrt_organ, element_list(el))
+       struct_m = ccohort%prt%GetState(struct_organ, element_list(el))
+       repro_m  = ccohort%prt%GetState(repro_organ, element_list(el))
+                
+       litt => cpatch%litter(el)
+       flux_diags => csite%flux_diags(el)
+
+       do c=1,ncwd
+                   
```

Comment:
add a comment flag for the speed readers like myself to slow me down and help me catch ag vs bg
```suggestion

! above ground CWD
```